docs/design/snapshot.md

```diff
@@ -0,0 +1,185 @@
+Snapshots are read-only copies of volumes (once the snapshot is activated) that
+can be used for cloning new volumes from. Its is possible to create a snapshot
+through the [Snapshot Create API](../api/api.md#create-a-snapshot) or the
+commandline client.
+
+From the command line client, you can type the following to create a snapshot
+from an existing volume:
+
+```
+$ heketi-cli volume snapshot-create <vol_uuid> [-name=<snap_name>]
+```
+
+The new snapshot can be used to create a new volume:
+
+```
+$ heketi-cli volume snapshot-clone <snap_uuid> [-name=<clone_name>]
+```
+
+The clones of snapshots are new volumes with the same properties as the
+original. The cloned volumes can be deleted with the `heketi-cli volume delete`
+command. In a similar fashion, snapshots can be removed with the `heketi-cli
+volume snapshot-delete` command.
+
+# Proposed CLI
+
+```
+$ heketi-cli volume snapshot-create <vol_uuid> [--name=<snap_uuid>] [--description=<string>]
+$ heketi-cli volume snapshot-clone <snap_uuid> [--name=<vol_uuid>]
+$ heketi-cli volume snapshot-delete <snap_uuid>
+$ heketi-cli volume snapshot-list <vol_uuid>
+$ heketi-cli volume snapshot-info <snap_uuid>
+```
+
+# API Proposal
+
+The API is layed out here for file volume types only. The samy API will be
+added for block-volumes at a later time.
+
+### Create a Snapshot
+* **Method:** _POST_
+* **Endpoint**:`/volumes/{volume_uuid}/snapshots`
+* **Content-Type**: `application/json`
+* **Response HTTP Status Code**: 202, See [Asynchronous Operations](#asynchronous-operations)
```

Comment:
I think this need not be an async call, any specific reason for making it async call?

Reply:
Gluster needs to do some work on different storage servers, and that may require some time. Doing this async seems appropriate?